### DISCUSSION TITLE
how do I resize layout and change percent position child (sprite) of layout or is there any way to change the child position when resizing the parent?

### DISCUSSION THREAD

**Discussion by hoafiaan** [axmolengine/axmol#2289] [General]
I use layout->setContentSize(Director::getInstance()->getVisibleSize()) and layout->forceDoLayout() but it doesn't working

**Comment by rh101** [comment]
Your requirements are not clear, but if they are somehow related to automatically adjusting child widget positions on a `ui::Layout`, then look into `Widget::setLayoutParameter`.  The cpp-tests has examples of layout parameter usage.

**Reply by hoafiaan** [reply]
I mean with child as sprite in ui::Layout

**Reply by rh101** [reply]
`ax::Sprite` does not inherit from `ax::Widget`, so it doesn't have the same layout capabilities.  

There are several ways around this, for example, use `ui::ImageView` if it's just a static image, or add the sprite as a child of a `ax::ui::Widget` or `ui::Layout`, and then that widget will be automatically positioned in the parent layout when it is resized.

You may also be able to sub-class Widget and add Sprite functionality to it, or inherit from both etc. etc..
